velox/exec/TopN.cpp

```diff
@@ -36,14 +36,30 @@ TopN::TopN(
           topNNode->sortingOrders(),
           data_.get()),
       topRows_(comparator_),
-      decodedVectors_(outputType_->children().size()) {}
+      decodedVectors_(outputType_->children().size()) {
+  sortingKeyColumns_.reserve(topNNode->sortingKeys().size());
+  const auto columnCount{outputType_->children().size()};
+  std::vector<bool> isSortingKey(columnCount);
+  for (const auto& key : topNNode->sortingKeys()) {
+    sortingKeyColumns_.emplace_back(exprToChannel(key.get(), outputType_));
+    isSortingKey[sortingKeyColumns_.back()] = true;
+  }
+  nonKeyColumns_.reserve(columnCount - sortingKeyColumns_.size());
```

Comment:
This logic assumes there are no duplicate sorting keys. This is a reasonable assumption, but it is not enforced. Let's update TopNNode constructor to enforce this.

Reply:
Added, thanks!